packages/jest-environment-winappdriver/src/WinAppDriverEnvironment.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @format
+ */
+
+import {spawn, ChildProcess} from 'child_process';
+import * as fs from 'fs';
+
+import NodeEnvironment = require('jest-environment-node');
+
+import * as webdriverio from 'webdriverio';
+import {BrowserObject, RemoteOptions} from 'webdriverio';
+import {Config} from '@jest/types';
+
+export type EnvironmentOptions = {
+  app?: string;
+  winAppDriverBin?: string;
+  webdriverOptions?: RemoteOptions;
+};
+
+class WinAppDriverEnvironment extends NodeEnvironment {
+  private readonly options: RemoteOptions;
+  private readonly winappdriverBin: string;
+  private winAppDriverProcess: ChildProcess | undefined;
+  private browser: BrowserObject | undefined;
+
+  constructor(config: Config.ProjectConfig) {
+    super(config);
+    const passedOptions: EnvironmentOptions = config.testEnvironmentOptions;
+
+    if (!passedOptions.app) {
+      throw new Error('"app" must be specified in testEnvironmentOptions');
+    }
+
+    this.winappdriverBin =
+      passedOptions.winAppDriverBin ||
+      'C:\\Program Files (x86)\\Windows Application Driver\\WinAppDriver.exe';
+
+    const baseOptions: RemoteOptions = {
+      port: 4723,
+      capabilities: {
+        app: passedOptions.app,
+        // @ts-ignore
+        'ms:experimental-webdriver': true,
+      },
+    };
+
+    this.options = Object.assign(
+      {},
+      baseOptions,
+      passedOptions.webdriverOptions,
+    );
+  }
+
+  async setup() {
+    await super.setup();
+
+    this.winAppDriverProcess = await spawnWinAppDriver(this.winappdriverBin);
+    this.browser = await webdriverio.remote(this.options);
+
+    this.global.remote = webdriverio.remote;
+    this.global.browser = this.browser;
+    this.global.$ = this.browser.$.bind(this.browser);
+    this.global.$$ = this.browser.$$.bind(this.browser);
+  }
+
+  async teardown() {
```

Comment:
Would it make sense to do the screenshot here, so it is avaialble to every winappdriver tests in the future rather than each test having to add it themsevles?

Reply:
Difference is moment of failure. This would take a screenshot after a suite is finished, and we couldn't tell if it fails. Using a reporter allows responding to a specific test failure at moment in time and taking a screenshot before moving on to the next test.